src/cargo/ops/cargo_test.rs

```diff
@@ -198,3 +201,24 @@ fn run_doc_tests(options: &TestOptions,
     }
     Ok((Test::Doc, errors))
 }
+
+fn find_host_deps(options: &TestOptions, compilation: &Compilation) -> OsString {
```

Comment:
Instead of being calculated here, could this perhaps be a field of the `Compilation` returned?